NitroxServer/GameLogic/Entities/EntityManager.cs

```diff
@@ -28,14 +28,46 @@ public class EntityManager
 
         public EntityManager(List<Entity> entities, BatchEntitySpawner batchEntitySpawner)
         {
-            entitiesById = entities.ToDictionary(entity => entity.Id);
+            entitiesById = new Dictionary<NitroxId, Entity>(entities.Count);
+            globalRootEntitiesById = new Dictionary<NitroxId, Entity>();
+            phasingEntitiesByAbsoluteCell = new Dictionary<AbsoluteEntityCell, List<Entity>>();
+            bool toWarn = false;
 
-            globalRootEntitiesById = entities.FindAll(entity => entity.ExistsInGlobalRoot)
-                                             .ToDictionary(entity => entity.Id);
-
-            phasingEntitiesByAbsoluteCell = entities.FindAll(entity => !entity.ExistsInGlobalRoot)
-                                                    .GroupBy(entity => entity.AbsoluteEntityCell)
-                                                    .ToDictionary(group => group.Key, group => group.ToList());
+            for (int i = 0; i < entities.Count; i++)
```

Comment:
Would a foreach not be better here as it would remove the index below and just loop without the counter?

Reply:
I considered that, and if it would be preferred that can be done, but generally for loop index access of List<T> is a bit faster than the enumeration of a foreach, and has near identical readability, so I tend to prefer it.